src/connectedk8s/azext_connectedk8s/_utils.py

```diff
@@ -86,24 +104,86 @@ def add_helm_repo(kube_config, kube_context):
         telemetry.set_exception(exception=e, fault_type=consts.Get_HelmRegistery_Path_Fault_Type,
                                 summary='Error while fetching helm chart registry path')
         raise CLIError("Error while fetching helm chart registry path: " + str(e))
-    if response.status_code == 200:
-        return response.json().get('repositoryPath')
-    telemetry.set_exception(exception=str(response.json()), fault_type=consts.Get_HelmRegistery_Path_Fault_Type,
-                            summary='Error while fetching helm chart registry path')
-    raise CLIError("Error while fetching helm chart registry path: {}".format(str(response.json())))
+    if r.content:
+        try:
+            return r.json().get('repositoryPath')
+        except Exception as e:
+            telemetry.set_exception(exception=e, fault_type=consts.Get_HelmRegistery_Path_Fault_Type,
+                                    summary='Error while fetching helm chart registry path')
+            raise CLIError("Error while fetching helm chart registry path from JSON response: " + str(e))
+    else:
+        telemetry.set_exception(exception='No content in response', fault_type=consts.Get_HelmRegistery_Path_Fault_Type,
+                                summary='No content in acr path response')
+        raise CLIError("No content was found in helm registry path response.")
+
+
+def arm_exception_handler(ex, fault_type, summary, return_if_not_found=False):
+    if isinstance(ex, AuthenticationError):
+        telemetry.set_user_fault()
+        telemetry.set_exception(exception=ex, fault_type=fault_type, summary=summary)
+        raise CLIError("Authentication error occured while making ARM request: " + str(ex) + "\nSummary: {}".format(summary))
+
+    if isinstance(ex, TokenExpiredError):
+        telemetry.set_user_fault()
+        telemetry.set_exception(exception=ex, fault_type=fault_type, summary=summary)
+        raise CLIError("Token expiration error occured while making ARM request: " + str(ex) + "\nSummary: {}".format(summary))
+
+    if isinstance(ex, HttpOperationError):
+        status_code = ex.response.status_code
+        if status_code == 404 and return_if_not_found:
+            return
+        if status_code // 100 == 4:
+            telemetry.set_user_fault()
+        telemetry.set_exception(exception=ex, fault_type=fault_type, summary=summary)
+        raise CLIError("Http operation error occured while making ARM request: " + str(ex) + "\nSummary: {}".format(summary))
+
+    if isinstance(ex, ValidationError):
+        telemetry.set_user_fault()
+        telemetry.set_exception(exception=ex, fault_type=fault_type, summary=summary)
+        try:
+            raise CLIError("Validation error occured while making ARM request: " + str(ex) + "\nSummary: {}".format(summary))
+        except AttributeError:
+            raise CLIError("Validation error occured while making ARM request: " + str(ex) + "\nSummary: {}".format(summary))
```

Comment:
why do you need the try except here when raising CLIError?

Reply:
Removed the try catch